File: 프로그래머스/unrated/140108. 문자열 나누기/문자열 나누기.kt
class Solution {
    fun solution(s: String): Int {
        var count = -1
        var answer = 0

        loop@ while (true) {
            val x = s[++count]
            var xCount = 1
            var notXCount = 0

            while (true) {
                if (count == s.length - 1) {
                    answer++
                    break@loop
                }

                count++

                if (x == s[count]) {
                    xCount++
                } else if (xCount == notXCount + 1) {
                    answer++
                    break
                } else {
                    notXCount++
                }
            }
            if (count == s.length - 1) break
        }
        return answer
    }
}

File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.swift
func solution(_ s:String) -> Int {
    var answer = 0
    var x = (name: "", count: 0)
    
    for s in s.map { String($0) } {
        if x.name == "" {
            x.name = s
            x.count += 1
        } else {
            x.count += (x.name == s) ? 1 : -1
            
            if x.count == 0 {
                answer += 1
                x = ("", 0)
            }
        }        
    }
    
    return (x.count != 0) ? answer + 1 : answer
}